src/main/java/stream/flarebot/flarebot/Events.java

```diff
@@ -367,6 +368,13 @@ public void onDisconnect(DisconnectEvent event) {
                     .getClientCloseFrame().getCloseReason()));
     }
 
+    @Override
+    public void onRoleDelete(RoleDeleteEvent event) {
+        if (FlareBotManager.getInstance().getSelfAssignRoles(event.getGuild().getId()).contains(event.getRole().getId())) {
+            FlareBotManager.getInstance().getSelfAssignRoles(event.getGuild().getId()).remove(event.getRole().getId());
```

Comment:
`FlareBotManager.getInstance().getSelfAssignRoles(event.getGuild().getId())`
Store it